bindings/rust/src/bindings/mod.rs

```diff
@@ -70,6 +42,10 @@ pub struct KZGProof {
     bytes: [u8; BYTES_PER_PROOF],
 }
 
+pub struct Blob {
+    bytes: Vec<u8>,
```

Comment:
I have a concern about these `Blob` types defined here and also in all the other bindings PRs.

The problem is that we need to manually length-check them everytime they are used. This manual length check was not done in the past because a Blob was always a blob, whereas now with the variable length, it's not necessary that a byte vector is a blob.

I feel like this use of the `Blob` type when you don't actually know that it's a blob (until you validate it with a length check) is a dangerous pattern. Maybe in these PRs we have performed all the length checks required, but I think it sets us in a bad state for future changes because a dev could reasonably think that an input Blob is a blob and forget to do the length check.

This also is an anti-pattern in our code base where we have established the (IMO good) `Bytes32`/`Bytes48` convention for pre-validation objects. Because now this new `Blob` type is pre-validated and it's still called Blob.

I think The Right Thing would have been to introduce a new Bytes type that can be turned into a Blob if it's valid (i.e. its size is right). Same way that a `Bytes32` can be turned into a `KZGCommitment` if it's valid.

(moved this from TG here so that it doesn't get lost)